### PR TITLE
Eredin classes and ai

### PR DESCRIPTION
**Pull request by PiotrEredin**

### CODE REVIEW

File: src/prefabs/creature.js [hunk 1]
```diff
@@ -58,9 +59,10 @@ class Creature extends Phaser.Sprite {
 
     update() {
         this.hpText.text = 'HP: ' + this.hp_curr + '/' + this.hp_max;
+        this.game.physics.arcade.collide(this, this.game.world.getByName('Platforms'));
```
Comment (Sulfastic) [inline src/prefabs/creature.js:62]: So the issue "walking into walls" was just to reorder the collision model?

File: src/prefabs/creature.js [hunk 2]
```diff
@@ -58,9 +59,10 @@ class Creature extends Phaser.Sprite {
...
         this.game.physics.arcade.collide(this, this.game.world.getByName('Players'));
         this.game.physics.arcade.collide(this, this.game.world.getByName('Enemies'));
-        this.game.physics.arcade.collide(this, this.game.world.getByName('Platforms'));
+        if(this.busy) this.stop();
```
Comment (Sulfastic) [inline src/prefabs/creature.js:65]: Please use {} even if it is one-liner - it really helps reading; in this case:
if(this.busy) {
    this.stop();
}

Comment (PiotrEredin) [reply src/prefabs/creature.js:65]: OK, will use {} like You asked.
Yes, just reordering was sufficient to make it work. Probably when the collision of the player with the enemy was called it pushed the creature through the platforms and then the collision with platforms didn't do anything.

Comment (Sulfastic) [reply src/prefabs/creature.js:65]: Although it's not only in this file, it may need small refactoring, so it should be made in separate task

**Review by Sulfastic — APPROVED** [review]

**Pull request merged**